File: problems/leetcode/1979-greatest_common_divisor_of_array.cpp
class Solution {
public:
    int findGCD(std::vector<int>& nums) {
        int smallest = *std::min_element(nums.begin(), nums.end());
        int highest = *std::max_element(nums.begin(), nums.end());

        return gcd(smallest, highest);
    }

private:
    int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
};

File: problems/leetcode/1979-greatest_common_divisor_of_array.java
class Solution {
    public int findGCD(int[] nums) {
        int smallest = nums[0];
        int highest = nums[0];
        int gcd = 1;

        for(int i = 0; i < nums.length; i++){
            if (nums[i] <= smallest){
                smallest = nums[i];
            }
            if (nums[i] > highest){
                highest = nums[i];
            }
        }
        for (int i = Math.min(smallest, highest); i > 0; i--){
            if (smallest % i == 0 && highest % i == 0){
                return i;
            }
        }
        return 1;
    }
}